x-pack/filebeat/module/microsoft/m365_defender/ingest/pipeline.yml

```diff
@@ -235,6 +235,25 @@ processors:
     field: url.full
     ignore_failure: true
     if: ctx?.url?.full != null
+
+  ######################
+  ## ECS Host Mapping ##
+  ######################
+- foreach:
+    field: json.alerts.devices
+    ignore_missing: true
+    processor:
+      append:
+        field: host.hostname
+        value: "{{ _ingest._value.deviceDnsName }}"
+        allow_duplicates: false
+        ignore_failure: true
```

Comment:
`host.hostname` is a keyword, so this is not correct. A possible approach is to perform a [split](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-httpjson.html#response-split) on `deviceDnsName` in the httpjson processors [here](https://github.com/elastic/beats/blob/master/x-pack/filebeat/module/microsoft/m365_defender/config/defender.yml).

Reply:
Are there any tests for the config in `defender.yml` ?

Reply:
The changes you have in m365_defender-test.ndjson.log{,-expected}.json are those.

Reply:
but none of those tests fail after making a change to `defender.yml`.  It would seem that any changes to the `defender.yml` would affect the way Filebeat (and httpjson) would run, which would change the *inputs* of those tests. But there doesn't seem to be any tests that actually run httpjson with the configuration provided, to convert raw events into those intermediary events used to test the Ingest Pipeline.

Reply:
Sorry, yes. I remember this being an issue for me when I was working on changes in httpjson, but forgot in my comment. Apologies. The inputs in the tests documents that would have come from httpjson, so changes in the httpjson will not have any impact. From memory, my approach to dealing with this was horribly hacky and so I won't recommend it. I will ask if there is a standard approach that _can_ be recommended.

Reply:
Yeah, i looked into copying e.g. `input_test.go` from httpjson, and loading the config from `defender.yml` ... but I'm not confident in my ability to get that working correctly.

Reply:
I suspect I did something like that.